packages/agreement-process/test/deleteAgreement.test.ts

```diff
@@ -98,6 +103,107 @@ describe("delete agreement", () => {
     ).not.toContain(agreement.consumerDocuments[1].path);
   });
 
+  it("should succeed when requester is Consumer Delegate and the Agreement is in a deletable state", async () => {
+    vi.spyOn(fileManager, "delete");
+    const agreementId = generateId<AgreementId>();
+    const consumerDocuments = [
+      getMockConsumerDocument(agreementId, "doc1"),
+      getMockConsumerDocument(agreementId, "doc2"),
+    ];
+
+    const agreement = {
+      ...getMockAgreement(),
+      id: agreementId,
+      state: randomArrayItem(agreementDeletableStates),
+      consumerDocuments,
+    };
+
+    const authData = getRandomAuthData(agreement.consumerId);
```

Comment:
Here you are saying that the requester is the consumer, not the delegate - you should create another tenant, the delegate, and use its id as the delegated and also as the id to pass here in authData, otherwise you are calling as the consumer, not as the delegate consumer